### PR TITLE
Red Hat Data Grid Last product enviorment workload

### PR DESCRIPTION
**Pull request by talaraujo**
##### SUMMARY

Creating a new workload to install Red Hat Data Grid operator and create a new empty cluster.

##### ISSUE TYPE
- New role Pull Request

##### COMPONENT NAME

Using install operator role to install Red Hat Data Grid  operator and created a new Task do configure a new Red Hat Data Grid cluster

##### ADDITIONAL INFORMATION
N/A

```paste below
ansible==2.9.27
argcomplete==1.12.3
awscli==1.22.60
Babel==2.9.1
bcrypt==3.2.0
Beaker==1.10.0
beautifulsoup4==4.9.3
blivet==3.4.3
blivet-gui==2.3.0
boto3==1.21.4
botocore==1.24.5
Brlapi==0.8.2
cffi==1.15.0
chardet==4.0.0
charset-normalizer==2.0.4
chrome-gnome-shell==0.0.0
click==8.0.1
colorama==0.4.3
configobj==5.0.6
cryptography==3.4.7
cupshelpers==1.0
dasbus==1.6
dbus-python==1.2.18
decorator==5.0.9
defusedxml==0.7.1
distro==1.6.0
docutils==0.15.2
fedora-third-party==0.8
fluidity-sm==0.2.0
fros==1.1
git-cola==3.11.0
gpg==1.15.1
httpie==3.1.0
humanize==0.5.1
idna==3.2
invoke==1.6.0
Jinja2==3.0.1
jmespath==0.10.0
langtable==0.0.57
lexicon==2.0.1
libcomps==0.1.18
lxml==4.6.5
Mako==1.1.4
MarkupSafe==2.0.0
mercurial==5.9.3
multidict==5.1.0
nftables==0.1
ntlm-auth==1.5.0
olefile==0.46
packaging==21.3
paramiko==2.9.1
Paste==3.5.0
pathspec==0.9.0
pexpect==4.8.0
pid==2.2.3
Pillow==8.3.2
ply==3.11
productmd==1.33
psutil==5.8.0
ptyprocess==0.6.0
pwquality==1.4.4
pyalsa==1.2.6
pyasn1==0.4.8
pycairo==1.20.1
pycparser==2.20
pycrypto==2.6.1
pycups==2.0.1
pycurl==7.44.1
pyenchant==3.2.2
Pygments==2.9.0
PyGObject==3.42.0
pyinotify==0.9.6
pykickstart==3.34
PyNaCl==1.4.0
pyOpenSSL==21.0.0
pyparsing==2.4.7
pyparted==3.12.0
PyQt5==5.15.6
PyQt5-sip==12.9.0
PySocks==1.7.1
python-augeas==1.1.0
python-dateutil==2.8.1
python-meh==0.50
pytz==2021.3
pyudev==0.22.0
pywinrm==0.4.1
pyxdg==0.27
PyYAML==5.4.1
QtPy==1.11.2
regex==2022.3.2
requests==2.27.0
requests-file==1.5.1
requests-ftp==0.3.1
requests-ntlm==1.1.0
requests-toolbelt==0.9.1
rpm==4.17.0
rsa==4.7.2
s3transfer==0.5.1
selinux==3.3
sepolicy==3.3
setools==4.4.0
simpleaudio==1.0.4
simpleline==1.8.2
six==1.16.0
slip==0.6.4
slip.dbus==0.6.4
sos==4.2
soupsieve==2.3.1
systemd-python==234
Tempita==0.5.2
terminator==2.1.1
urllib3==1.26.7
xmltodict==0.12.0
yamllint==1.26.3


```

### CODE REVIEW

File: ansible/roles_ocp_workloads/ocp4_workload_lpe_datagrid/templates/datagrid-secret.yaml.j2 [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+type: Opaque
+stringData:
+  identities.yaml: |-
+    credentials:
+    - username: developer
+      password: developer
+    - username: user01
+      password: user01
+    - username: mysecondusername
+      password: changeme-two
```
Comment (wkulhanek) [inline ansible/roles_ocp_workloads/ocp4_workload_lpe_datagrid/templates/datagrid-secret.yaml.j2:16]: Passwords I would define in the defaults/main.yaml using a variable. And then set them in the AgnosticV config as you're deploying.

**Review by wkulhanek — CHANGES_REQUESTED** [review]

File: ansible/roles_ocp_workloads/ocp4_workload_lpe_datagrid/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+# Catalog Source Name when using a catalog snapshot. This should be unique
+# in the cluster to avoid clashes
+ocp4_workload_lpe_datagrid_catalogsource_name: redhat-operators-snapshot-datagrid
+
+# Catalog snapshot image
+ocp4_workload_lpe_datagrid_catalog_snapshot_image: quay.io/gpte-devops-automation/olm_snapshot_redhat_catalog
+
+# Catalog snapshot image tag
+ocp4_workload_lpe_datagrid_catalog_snapshot_image_tag: "v4.7_2021_04_12"
```
[the remark this answers is not in the document]
Comment (talaraujo) [reply ansible/roles_ocp_workloads/ocp4_workload_lpe_datagrid/defaults/main.yml]: changed the version to v4.9_2022_04_25

File: ansible/roles_ocp_workloads/ocp4_workload_lpe_datagrid/tasks/workload.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+# Create the AMQ Streams Platform
+- name: "Create the Data Grid Platform"
+  kubernetes.core.k8s:
+    state: present
+    merge_type:
+      - strategic-merge
+      - merge
+    definition: "{{ lookup('template', item ) | from_yaml }}"
+  loop:
+    - ./../templates/datagrid-secret.yaml.j2
```
[the remark this answers is not in the document]
Comment (talaraujo) [reply ansible/roles_ocp_workloads/ocp4_workload_lpe_datagrid/tasks/workload.yml]: removed template path for files

File: ansible/roles_ocp_workloads/ocp4_workload_lpe_datagrid/templates/datagrid-cluster.yaml.j2 [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    #   org.infinispan: debug
+    #   org.jgroups: debug
+  replicas: 3
+  security:
+    endpointAuthentication: true
+    endpointSecretName: {{ ocp_username }}-credentials
+  service:
+    container:
+      ephemeralStorage: true
+    type: DataGrid
```
[the remark this answers is not in the document]
Comment (talaraujo) [reply ansible/roles_ocp_workloads/ocp4_workload_lpe_datagrid/templates/datagrid-cluster.yaml.j2]: added a newline

File: ansible/roles_ocp_workloads/ocp4_workload_lpe_datagrid/templates/datagrid-secret.yaml.j2 (hunk 1 above)
Comment (talaraujo) [reply ansible/roles_ocp_workloads/ocp4_workload_lpe_datagrid/templates/datagrid-secret.yaml.j2]: removed and added it in defaults main file

**Review by wkulhanek — APPROVED** [review]

**Pull request merged**